### PR TITLE
[RDBMS] `az postgres flexible-server geo-restore/replica`: Introduce read replicas and geo-restore

### PR DESCRIPTION
**Pull request by alanenriqueo**
**Related command**
- `az postgres flexible-server geo-restore`
- `az postgres flexible-server replica create`
- `az postgres flexible-server replica list`
- `az postgres flexible-server replica stop-replication`

**Description**
Introduce replicas and geo-restore for PostgreSQL flexible server:
- Replica location can be different from primary server location.
- Restored and geo-restored servers can decide whether to be geo-redundant or not.
- From a server with public access, we can only:
   - create a replica with public access.
   - geo-restore with public access
- From a server with vnet, we can only:
   - create a replica with a vnet in same region.
   - geo-restore with a vnet in paired region.

Changes in code:
- Add `geo_redundant_backup` argument in functions `flexible_server_create` and `flexible_server_restore`.
- Add functions `flexible_server_georestore`, `flexible_replica_create`, `flexible_replica_stop`.
- Refactor vnet, subnet and private dns zone provisionings in method `flexible_server_provision_network_resource`.
- Update azure-mgmt-rdbms to version 10.2.0b5.

**Testing Guide**

See examples in _helptext_pg.py and updated tests.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/_util.py [hunk 1]
```diff
@@ -84,3 +85,22 @@ def create_firewall_rule(cmd, resource_group_name, server_name, start_ip, end_ip
...
+
+def retryable_method(retries=3, interval_sec=5, exception_type=Exception, condition=None):
+    def decorate(func):
+        def call(*args, **kwargs):
+            current_retry = retries
+            while True:
+                try:
+                    return func(*args, **kwargs)
+                except exception_type as ex:  # pylint: disable=broad-except
+                    print(condition(ex))
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_util.py:98]: Is this `print` for debug purpose?

Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/_util.py:98]: Ups, removing it.

File: src/azure-cli/azure/cli/command_modules/rdbms/_util.py [hunk 2]
```diff
@@ -84,3 +85,21 @@ def create_firewall_rule(cmd, resource_group_name, server_name, start_ip, end_ip
     firewall = firewall_client.begin_create_or_update(resource_group_name, server_name, firewall_name, parameters)
 
     return firewall.result().name
+
+
+def retryable_method(retries=3, interval_sec=5, exception_type=Exception, condition=None):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_util.py:90]: We have retry decorator in cli core: https://github.com/Azure/azure-cli/blob/7b12d40626bd52f60ef601c727d5d8b9433bc886/src/azure-cli-core/azure/cli/core/decorators.py#L89-L113

Maybe we can add `condition` to cli core retry decorator in the future. @jiasli for awareness

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/rdbms/_util.py (hunk 2 above)
Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/_util.py:90]: Yeah, I noticed that decorator before, but I want to check a specific property of the exception to know if it's retryable or not.

**Pull request merged**